rotkehlchen/globaldb/handler.py

```diff
@@ -1634,54 +1636,56 @@ def soft_reset_assets_list() -> tuple[bool, str]:
         builtin_database = root_dir / 'data' / 'global.db'
         detach_database = 'DETACH DATABASE "clean_db";'
 
-        try:
-            with GlobalDBHandler().conn.read_ctx() as read_cursor:
-                read_cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
-                # Check that versions match
-                query = read_cursor.execute('SELECT value from clean_db.settings WHERE name="version";')  # noqa: E501
-                version = query.fetchone()
-                if version is None or int(version[0]) != globaldb_get_setting_value(read_cursor, 'version', GLOBAL_DB_VERSION):  # noqa: E501
+        with GlobalDBHandler().packaged_db_lock:
+            try:
+                with GlobalDBHandler().conn.read_ctx() as read_cursor:
+                    read_cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
+                    # Check that versions match
+                    query = read_cursor.execute('SELECT value from clean_db.settings WHERE name="version";')  # noqa: E501
+                    version = query.fetchone()
+                    if version is None or int(version[0]) != globaldb_get_setting_value(read_cursor, 'version', GLOBAL_DB_VERSION):  # noqa: E501
+                        read_cursor.execute(detach_database)
+                        msg = (
+                            'Failed to restore assets. Global database is not '
+                            'updated to the latest version'
+                        )
+                        return False, msg
+
+                    # Get the list of ids that we will restore
+                    query = read_cursor.execute('SELECT identifier from clean_db.assets;')
+                    shipped_asset_ids = set(query.fetchall())
```

Comment:
you didn't write it but I believe is possible to do just `set(query)` isn't it?

Reply:
Should be